lib/marked.js

```diff
@@ -539,7 +539,7 @@ inline.pedantic = merge({}, inline.normal, {
 
 inline.gfm = merge({}, inline.normal, {
   escape: replace(inline.escape)('])', '~|])')(),
-  url: XRegExp.cache('^((?:[A-Za-z][\\pL\\d-]+([.+][A-Za-z\\d]+)?:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{};:\'",<>?«»“”‘’]|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
+  url: XRegExp.cache('^((?:[A-Za-z][A-Za-z\\d-.+]*[A-Za-z\\d]+:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{};:\'",<>?«»“”‘’]|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
```

Comment:
@lieut-data  also a scheme should end with the alpha or num so here is the final version of the regex according to your suggestion.

Reply:
Thanks for the update, @letsila -- can you point me to the requirement that a scheme ends with an alphanumeric character? I know hosts have a similar limitation, but the linked RFC only states:
```
Scheme names consist of a sequence of characters beginning with a
   lower case letter and followed by any combination of lower case
   letters, digits, plus ("+"), period ("."), or hyphen ("-").  For
   resiliency, programs interpreting URI should treat upper case letters
   as equivalent to lower case in scheme names (e.g., allow "HTTP" as
   well as "http").

      scheme        = alpha *( alpha | digit | "+" | "-" | "." )
```

Reply:
The official list of available format for schemes can be retrieved from this page https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml and none of the format inside this [document](https://www.iana.org/assignments/uri-schemes/uri-schemes-1.csv) show something like `http.://` or `http+://`.

In addition to that a previous [test case](https://github.com/mattermost/marked/blob/master/test/tests/test-links.text#L101) specifies that `test.:test` should not render as a link.

Reply:
@lieut-data 👆🏽did that answer your question?

Reply:
@letsila, it may very well be that we want to deviate from what is a valid scheme for our own purposes, but I think the RFC is authoritative here and that it allows for `test.:test` (even if no one has defined such a scheme before). I also point -- non-authoritatively -- to the [url parsing](https://golang.org/src/net/url/url.go#L435) library in Go which makes no provision for trying to ignore such characters at the end of the scheme.

@hmhealey, is there a Mattermost-specific reason we should reject this kind of scheme? Otherwise, I propose that we stick to the RFC.

As an aside, I believe the current proposal also enforces a minimum of two-characters in the scheme, which is also not required by the RFC. Solving that would require lookaheads, I think.

Reply:
@lieut-data make sense. I've just updated the regex according to the RFC :)